pkg/proxy/iptables/proxier.go

```diff
@@ -791,11 +798,25 @@ func (proxier *Proxier) execConntrackTool(parameters ...string) error {
 // assumes proxier.mu is held
 func (proxier *Proxier) syncProxyRules() {
 	if proxier.throttle != nil {
-		proxier.throttle.Accept()
+		// NOTE: using non-blocking TryAccept.
+		// All other updates should update the datastructure
+		// and do the appropriate thing to batch when we are not syncing.
+		// Lastly, the sync timer will be set to ensure we catch the last update.
+		if proxier.throttle.TryAccept() == false {
+			duration := time.Since(proxier.lastSync)
+			if duration < proxier.minSyncPeriod {
+				glog.V(4).Infof("Attempting to sych too often.  Duration: %v, min period: %v", duration, proxier.minSyncPeriod)
+				proxier.timer.Reset(proxier.minSyncPeriod - duration)
```

Comment:
If we hit this, then return, `Sync()` just resets it to the longer timer, doesn't it?